platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Style.java

```diff
@@ -156,6 +194,11 @@ public boolean removeSource(@NonNull Source source) {
    * @param layer the layer to add
    */
   public void addLayer(@NonNull Layer layer) {
+    if (!fullyLoaded) {
+      // we are loading a new style
+      return;
```

Comment:
How about throwing an exception in `add` cases? Adding a component to a style that is already overwritten by a new one is an error that's going to be hidden from the end user and require more debugging than necessary.

I'm on board with returning `null` or empty objects for getters though.

Reply:
> I'm on board with returning null or empty objects for getters though.

On the second thought, not sure if I can see a valid scenario of this situation either.